Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/NewMeterWizard/AssetPage.tsx

```diff
@@ -275,6 +275,12 @@ export default function AssetPage(props: IProps) {
             return <StationBatteryAttributes NewEdit={newEdit} Asset={newEditAsset} UpdateState={setNewEditAsset} />;
     }
 
+    function filterChannels(chs: OpenXDA.Types.Channel[]) {
+        return chs.filter(ch =>
+            !(ch.Asset.length > 0)
+        );
+    }
```

Comment:
Can this be wrapped with a useMemo instead of being a function? That way, we only run this function once any time props.Channels changes and not a bunch of times every rerender.

Reply:
I'm not sure that's true. I don't think we run this function on every rerender. It just get's defined as a new function on every re-render. that may be perfectly fine. 
But the real question is why are we filtering through this twice? 
`ChannelSelector` will filter it by whatever matches the Search state, so it will run through the list anyway. It would make sense to add this Filter there instead of filtering it here and in `ChannelSelector`